src/components/reference/implementation/fission-base.ts

```diff
@@ -16,7 +16,13 @@ export async function implementation(endpoints: Endpoints, dependencies: Depende
   base.dataRoot.domain = (username: string) => `${username}.files.${endpoints.userDomain}`
   base.dataRoot.lookup = (...args) => DataRoot.lookup(endpoints, dependencies, ...args)
   base.dataRoot.update = (...args) => DataRoot.update(endpoints, dependencies, ...args)
-  base.didRoot.lookup = (...args) => DID.root(endpoints, ...args)
+  base.didRoot.lookup = (...args) => {
+    try {
+      return DID.root(endpoints, ...args)
+    } catch {
+      return Base.didRootLookup(dependencies, ...args)
```

Comment:
Some context, recently I had an issue that the root DID in my UCANs was different from the one in DNS. I assume because of account recovery or an issue with the CLI (it's very old account).

In any case, I'm not sure about this approach. I do know that the DID returned from this function should always be the same, otherwise I introduce a bunch of issues, such as problems loading the file system. So ideally I think we pick one or the other, not the above approach where it would use the local-only approach (Base) if the DNS lookup fails (eg. when online or DNS services are down).

I would opt for the local-only approach, but then we have the new UCAN spec that no longer uses nested UCANs in the way we use them now (will be referenced with CIDs). So I'm not sure in how far that will work offline.

Thoughts @bgins @matheus23 ?

Reply:
Will default to DNS for now.